platform/openide.util/src/org/netbeans/modules/openide/util/NbBundleProcessor.java

```diff
@@ -309,8 +311,12 @@ private String findCompilationUnitName(Element e) {
         case INTERFACE:
         case ENUM:
         case ANNOTATION_TYPE:
-            switch (e.getEnclosingElement().getKind()) {
-            case PACKAGE:
+            if (e.getEnclosingElement().getKind() == PACKAGE) {
+                return e.getSimpleName().toString();
+            }
+        }
+        if ("RECORD".equals(e.getKind().name())) {
```

Comment:
I guess the string matching is due to future compatibility, is it?
The indentation of the switch case block is bad, but I guess that how it was before this change, might fix that for the readability.

Reply:
The build fails without it since platform builds against 8 without nb-javac in the classpath.

Java modules can "cheat" and use the RECORD enum directly since nb-javac adds it to the classpath due to the fact that it is currently based on javac of JDK 19.

So its technically about backwards compatibility with JDK 8.

Reply:
updated the formatting of the switch - I agree that this was confusing. Also added a comment.